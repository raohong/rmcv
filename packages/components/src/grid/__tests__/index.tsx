import { render, screen } from '@testing-library/react';
import { Grid } from '..';

const testId = 'grid';

it('render correctly', () => {
  const tree = render(<Grid />);

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with column', () => {
  const column = 3;

  render(<Grid data-testid={testId} column={column} />);
  expect(screen.getByTestId(testId)).toHaveStyle({
    gridTemplateColumns: `repeat(${column}, minmax(0, 1fr))`,
  });
});

it('render with gutter', () => {
  const gutter = 16;

  render(<Grid data-testid={testId} gutter={gutter} />);
  expect(screen.getByTestId(testId)).toHaveStyle({
    gap: `${gutter}px`,
  });
});
