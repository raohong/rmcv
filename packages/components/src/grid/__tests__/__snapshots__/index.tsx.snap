// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
  display: grid;
  gap: 0;
  grid-template-columns: repeat(4, minmax(0, 1fr));
}

.emotion-0::after {
  content: "";
  position: absolute;
  left: -50%;
  top: -50%;
  bottom: -50%;
  right: -50%;
  border-style: solid;
  border-width: 1px;
  -webkit-transform: scale(0.5);
  -moz-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
  pointer-events: none;
  border-color: #ebedf0;
}

<div
    class="RVuiGrid-root RVuiGrid-vertical emotion-0"
  />
</DocumentFragment>
`;
